app/assets/stylesheets/pages/nodes_list.scss

```diff
@@ -0,0 +1,28 @@
+.discovery-control {
```

Comment:
I think this MR is already a monster, we can reorganize later on extra PR's, unless there's something "blocking".